tools/qlog/src/lib.rs

```diff
@@ -1148,6 +1465,37 @@ impl PacketHeader {
             dcid,
         }
     }
+
+    /// Creates a new PacketHeader.
+    ///
+    /// Once a QUIC connection has formed, version, dcid and scid are stable, so
+    /// there are space benefits to not logging them in every packet, especially
+    /// PacketType::OneRtt.
+    pub fn with_type(
+        ty: PacketType, packet_number: u64, packet_size: Option<u64>,
+        payload_length: Option<u64>, version: Option<u32>, scid: Option<&[u8]>,
```

Comment:
I think packet_size and payload_length is good enough with `usize` since it's all udp payload? Any reason to use u64 here?

Reply:
The qlog schema defines this field type as `number` - see https://quiclog.github.io/internet-drafts/draft-marx-qlog-event-definitions-quic-h3.html#name-notational-conventions

When starting off, I took the very simple option of declaring any field type of `number` as u64. It wouldn't hurt to analyse all of these and figure out if we can shrink them.

Reply:
But also there is a bit of activity right now in qlog to rationalise the logging of numerical fields, which are sometimes written as strings.

Reply:
Yes, and it looks like draft-02 will say something like:

> just use numbers but be aware that some platforms (e.g., web-based tools) cannot cope with numbers > 2^53, so if you have -very- long running logs or start at a very high packet number offset, you can run into troubles. (note that this is a current limitation of the webbrowsers' JavaScript engines, not the JSON format itself)

For things like packets and QUIC-level frame lengths, 64-bit probably isn't needed. But in JSON/TypeScript there is only 1 "number" type. If we ever move qlog to a different schema format than TypeScript, we might be more specific in what's expected. 

In short: feel free to strip out numbers-as-strings and encode things as small as they need to be here.

Reply:
This PR is already pretty gargantuan. I'm going to do this in a separate PR that probably updates the qlog crate to draft -02 and bumps its version again.